src/blocks/sign-in/sign-in.tsx

```diff
@@ -66,13 +69,48 @@ export function SignIn({ title, subtitle }: SignInProps): JSX.Element {
             <Link href="/terms-of-service">Terms of Service</Link> and{' '}
             <Link href="/privacy-policy">Privacy Policy</Link>.
           </Text>
+          <CredentialsSignInForm />
         </div>
       </div>
       <div css={[tw`flex-1 hidden sm:block`, bannerStyle]}></div>
     </div>
   );
 }
 
+function CredentialsSignInForm(): JSX.Element {
+  const { register, handleSubmit } = useForm({
+    defaultValues: {
+      username: '',
+      password: '',
+    },
+  });
+  const [showForm, setShowForm] = React.useState(false);
+  React.useEffect(() => {
+    if (['staging', 'preview', 'localhost'].includes(getHostEnv())) {
+      setShowForm(true);
+    }
+  }, []);
+  const handleClickSubmit = handleSubmit(async (fields) => {
+    await signIn('credentials', {
+      // redirect: false,
+      callbackUrl: '/dashboard',
+      ...fields,
+    });
+  });
+  if (!showForm) {
+    return <></>;
+  }
+  return (
+    <div>
+      <TextField {...register('username')} type="text" label="Username" tw="w-full"></TextField>
+      <TextField {...register('password')} type="password" label="Password" tw="w-full"></TextField>
```

Comment:
I think <TextField {...register('password')} type="password" label="Password" tw="w-full" /> is better

Reply:
Sure